aws/resource_aws_codebuild_webhook.go

```diff
@@ -0,0 +1,112 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codebuild"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCodeBuildWebhook() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCodeBuildWebhookCreate,
+		Read:   resourceAwsCodeBuildWebhookRead,
+		Delete: resourceAwsCodeBuildWebhookDelete,
+		Update: resourceAwsCodeBuildWebhookUpdate,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"branch_filter": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCodeBuildWebhookCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codebuildconn
+
+	resp, err := conn.CreateWebhook(&codebuild.CreateWebhookInput{
+		ProjectName:  aws.String(d.Get("name").(string)),
+		BranchFilter: aws.String(d.Get("branch_filter").(string)),
+	})
+	if err != nil {
+		return err
+	}
+
+	d.SetId(d.Get("name").(string))
+	d.Set("branch_filter", d.Get("branch_filter").(string))
```

Comment:
`d.Set()` should only be called in the read function 😄 -- will fix this on merge.